### PR TITLE
update for 0.6

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

**Comment by KristofferC** [comment]
I'm gonna drop 0.4 support since for the `.` function calls.

File: src/generic.jl [hunk 1]
```diff
@@ -117,18 +117,12 @@ end
 function pairwise(metric::PreMetric, a::AbstractMatrix, b::AbstractMatrix)
     m = size(a, 2)
     n = size(b, 2)
-    r = Array(result_type(metric, a, b), (m, n))
+    r = Matrix{result_type(metric, a, b)}(m, n)
     pairwise!(r, metric, a, b)
 end
 
-function pairwise(metric::PreMetric, a::AbstractMatrix)
+function pairwise(metric::Union{PreMetric, SemiMetric}, a::AbstractMatrix)
```
Comment (timholy) [inline src/generic.jl]:
```julia
julia> supertype(SemiMetric)
Distances.PreMetric                                                                                                                    
```

File: test/REQUIRE [hunk 2]
```diff
@@ -1 +1 @@
-BaseTestNext
+julia 0.5
```
Comment (timholy) [inline test/REQUIRE]: wrong REQUIRE file

Comment (KristofferC) [inline test/REQUIRE]: woops!

**Comment by timholy** [comment]
Really nice work, @KristofferC.

**Comment by KristofferC** [comment]
Thanks, good to go from my point of view? Mac workers are sleeping as usual. Any more comments?

**Comment by timholy** [comment]
LGTM.

**Comment by KristofferC** [comment]
I wonder if attobot is activated here.. I'll try

**Comment by KristofferC** [comment]
Yep, cool: https://github.com/JuliaLang/METADATA.jl/pull/7864

**Pull request merged**